linux-glibc-2.17-x64/Dockerfile

```diff
@@ -0,0 +1,117 @@
+FROM ubuntu:24.04
+
+ARG GO_VERSION
+ARG GO_SHA256_LINUX_AMD64
+ARG CTNG_VERSION=1.26.0
+ARG IBM_MQ_VERSION=9.2.4.0
+ARG IBM_MQ_SHA256="d0d583eba72daf20b3762976f8831c2e23150ace90509520e12f8cda5b5bdb49"
+ARG RUST_VERSION=1.76.0
+ARG RUSTC_SHA256="0b2f6c8f85a3d02fde2efc0ced4657869d73fccfce59defb4e8d29233116e6db"
+ARG RUSTUP_VERSION=1.26.0
+ARG RUSTUP_SHA256="0b2f6c8f85a3d02fde2efc0ced4657869d73fccfce59defb4e8d29233116e6db"
+ARG BUNDLER_VERSION=2.4.20
+ARG DD_TARGET_ARCH
+
+ENV CONDA_PATH /root/miniforge3
+
+# Remove the early return on non-interactive shells, which makes sourcing the file not activate conda
+RUN grep -v return /root/.bashrc >> /root/newbashrc && cp /root/newbashrc /root/.bashrc
+
+RUN apt update -qy && apt install -y \
+    wget xz-utils gpg build-essential flex texinfo unzip \
+    help2man file gawk libtool-bin bison libncurses-dev \
+    python-is-python3 git cmake curl fakeroot procps bzip2 \
+    pkg-config libssl-dev libcurl4-openssl-dev libexpat-dev libpq-dev libz-dev \
+    rpm tar gettext autopoint autoconf clang libtool-bin \
+    pkg-config flex meson selinux-basics squashfs-tools gpg xz-utils gnupg2 patchelf cpio
+
+COPY linux-glibc-2.17-x64/config-x86_64-unknown-gnu-linux-glibc2.17 /build/crosstool-ng-${CTNG_VERSION}/.config
+COPY linux-glibc-2.17-x64/toolchain.cmake /opt/cmake/x86_64-unknown-linux-gnu.toolchain.cmake
+COPY linux-glibc-2.17-x64/config.toml /root/.cargo/config.toml
+
+RUN git config --global user.email "package@datadoghq.com"
+RUN git config --global user.name "Bits"
+
+# Build texinfo to get a bugfix version
+# Otherwise generating glibc doc with a recent version
+# causes a segfault. The fix isn't yet released as a package
+# https://git.savannah.gnu.org/cgit/texinfo.git/commit/?id=60d3edc4b74b4e1e5ef55e53de394d3b65506c47
+RUN cd /build && \
+    git clone --depth=1 https://git.savannah.gnu.org/git/texinfo.git && \
+    cd texinfo && \
+    ./autogen.sh && \
+    ./configure && make -j$(ncproc) && make install && \
+    rm -rf /build/texinfo
+
+RUN cd /build && \
+    wget http://crosstool-ng.org/download/crosstool-ng/crosstool-ng-${CTNG_VERSION}.tar.xz && \
+    gpg --keyserver pgp.surfnet.nl --recv-keys 1F30EF2E && \
+    wget http://crosstool-ng.org/download/crosstool-ng/crosstool-ng-${CTNG_VERSION}.tar.xz.sig && \
+    gpg --verify crosstool-ng-${CTNG_VERSION}.tar.xz.sig && \
+    tar xf crosstool-ng-${CTNG_VERSION}.tar.xz && \
+    cd /build/crosstool-ng-${CTNG_VERSION} && \
+    ./configure --enable-local && make -j$(nproc) && \
+    export CT_ALLOW_BUILD_AS_ROOT_SURE=yes && \
```

Comment:
if we need that set, should we set outside the RUN command or unset later?

Reply:
It only matters while building the toolchain, not when using it, so it doesn't need to be set outside of this specific command

Reply:
🤔 would an `export`ed variable get included in the final image? I'm not sure off the top of my head.

Reply:
No, this variable will stop existing outside of this specific layer